pkg/registry/client.go

```diff
@@ -117,9 +118,20 @@ func NewClient(options ...ClientOption) (*Client, error) {
 		}
 
 		// if username and password are set, use them for authentication
-		// by adding the basic auth Authorization header to the resolver
-		if client.username != "" && client.password != "" {
-			concat := client.username + ":" + client.password
+		username := client.username
+		password := client.password
+
+		// If credentials are not explicitly set, try to get them from .netrc
+		if username == "" && password == "" {
+			if creds, err := netrc.GetCredentials(ref.Registry); err == nil && creds != nil {
```

Comment:
Above we have `client.authorizer`, whose purpose is to abstract away the method of retrieving credentials. Then we have added `netrc` implementation directly here.

I'm not exactly sure how we can support `netrc`, as the `authorizer` mechanism is fairly tied to OCI credentials. But I don't think we should add `netrc` implementation directy into here